os/kernel/binary_manager/binary_manager_binfile.c

```diff
@@ -126,6 +139,80 @@ void binary_manager_scan_ubin(void)
 	}
 }
 
+/****************************************************************************
+ * Name: binary_manager_scan_ubin
+ *
+ * Description:
+ *	 This function scans user binary files.
+ *
+ ****************************************************************************/
+int binary_manager_scan_ubin(int bin_idx)
+{
+	int ret;
+	DIR *dirp;
+	int file_idx;
+	int name_len;
+	int version;
+	int latest_ver;
+	int latest_idx;
+	char *bin_name;
+	binary_header_t header_data;
+	char filepath[FILES_PER_BIN][CONFIG_PATH_MAX];
+
+	bmvdbg("Open a directory, %s\n", BINARY_DIR_PATH);
+
+	/* Open a directory for user binaries, BINARY_DIR_PATH */
+	dirp = opendir(BINARY_DIR_PATH);
+	if (!dirp) {
+		bmdbg("Failed to open a directory, %s\n", BINARY_DIR_PATH);
+		return ERROR;
+	}
+
+	file_idx = 0;
+	latest_ver = -1;
+	latest_idx = -1;
+	bin_name = BIN_NAME(bin_idx);
+	name_len = strlen(bin_name);
+
+	BIN_FILECNT(bin_idx) = 0;
+
+	/* Read each directory entry */
+	for (;;) {
+		struct dirent *entryp = readdir(dirp);
+		if (!entryp) {
+			/* Finished with this directory */
+			break;
+		}
+		/* Read and Verify a binary file */
+		if (DIRENT_ISFILE(entryp->d_type) \
+			&& !strncmp(entryp->d_name, bin_name, name_len) && entryp->d_name[name_len] == '_') {
+			snprintf(filepath, CONFIG_PATH_MAX, "%s/%s", BINARY_DIR_PATH, entryp->d_name);
+			ret = binary_manager_read_header(filepath, &header_data, false);
+			if (ret == OK) {
+				BIN_FILECNT(bin_idx)++;
+				strncpy(BIN_FILEVER(bin_idx, file_idx), header_data.bin_ver, BIN_VER_MAX);
+				version = (int)atoi(header_data.bin_ver);
+				bmvdbg("Found valid header in version %d\n", version);
```

Comment:
I think `Found valid header with version` is better. Because header is not in version.